objects/obj_player.object.gmx

```diff
@@ -273,13 +289,18 @@ if (vsp &lt; maxfallspeed) vsp += grav; // Increase downward movement speed base
 // Allow jumping if on floor, but don't allow it if on a platform, crouching, and not moving or while dash attacking
 if (onground &amp;&amp; !(place_meeting(x, y+1, obj_platform) &amp;&amp; hsp == 0 &amp;&amp; down_pressed) &amp;&amp; dashAttackTimer == dashAttackCool)
 {
-    vsp = jump_pressed * -jumpspeed;
     if(running &amp;&amp; jump_pressed){ // check if running when jumping
         sprite_index = spr_player_runjump; //change to runjump sprite
         run_jump = true; 
     }
 }
 
+if jump_pressed &amp;&amp; canJump // Player can only jump when the jump is pressed and the player is allowed to jump

```

Comment:
In doing this you have effectively nullified all the checks done above. This has brought back the ability to cancel a dash attack with a jump and other such bugs. instead of creating a whole new if statement, simply replace `onground` with `canJump` in the original if statement and add the `canJump = false;` line as well.